### PR TITLE
Add the launcher and some more mocks.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: Bootloader/firmware/src/launcher.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#endif
+
+/**
+ * @{
+ * @file launcher.h
+ * @brief Jump to the main application.
+ */
+
+/**
+ * @brief Jump to the main application at the specificed address.
```
Comment (peternewman) [inline Bootloader/firmware/src/launcher.h]: SPaG specified

Comment (nomis52) [inline Bootloader/firmware/src/launcher.h]: Fixed.

**Comment by peternewman** [comment]
LGTM

**Comment by peternewman** [comment]
LGTM

**Pull request merged**